.github/workflows/merge.yml

```diff
@@ -121,6 +121,43 @@ jobs:
           # we do many more runs on the nightly run
           PROPTEST_CASES: 50 
 
+  wasm:
```

Comment:
We want them run on `merge` to ensure that the code compiles. Any failure there should reject the PR.

Reply:
I mean move them into a separate workflow file, similar to `memcheck.yml`
will still be run per PR and block merge if any error happens.

the `merg.yml` to me shall just contain tests that involves function/feature related tests.
the naming of `merge` just because it is the first workflow file we have,
and may better to be renamed.
it does now contain too many separate sets of runs as I feel.